package/MDAnalysis/coordinates/chain.py

```diff
@@ -56,13 +56,124 @@
 from . import core
 
 
+def multi_level_argsort(l):
+    """Return indices to sort a multi value tuple. Sorting is done on the first
+    value of the tuple.
+
+
+    Parameters
+    ----------
+    l : list
+
+    Returns
+    -------
+    indices
+
+    Example
+    -------
+    >>> multi_level_argsort(((0, 2), (4, 9), (0, 4), (7, 9)))
+    [0, 2, 1, 3]
+
+    """
+    return [el[0] for el in sorted(enumerate(l), key=lambda x: x[1][0])]
+
+
+def filter_times(times, dt):
+    """Given a list of start and end times this function filters out any duplicate
+    time steps preferring the last tuple.
+
+    Parameters
+    ----------
+    times : list
+        sorted list of times
+    dt : float
+        timestep between two frames
+
+    Returns
+    -------
+    list:
+        indices of times to be used with overlaps removed
+
+    Example
+    -------
+    >>> filter_times(((0, 3), (0, 3)))
+    [1, ]
+    >>> filter_times(((0, 3), (0, 4)))
+    [1, ]
+    >>> filter_times(((0, 3), (3, 4)))
+    [0, 1]
+    >>> filter_times(((0, 3), (2, 5), (4, 9)))
+    [1, 2, 3]
+
+    """
+    # Special cases
+    if len(times) == 1:
+        return [0, ]
+    elif len(times) == 2:
+        if times[0][0] < times[1][0]:
+            return [0, 1]
+        elif np.allclose(times[0][0], times[1][0]):
+            return [1, ]
+        else:
+            return [0, ]
+    if np.unique(times).size == 2:
+        return [len(times) - 1, ]
+
+    # more then 2 unique time entries
```

Comment:
I forgot to remove that comment.

Reply:
That was not the comment I thought it was. This one should stay.

Reply:
anything you still want me to change?

Reply:
I wanted to remove that comment because out of context I thought it was about the couple of lines I removed there. But it actually applies to what follow so it should stay. I am happy with the PR.